### ISSUE TITLE
Remove all compiler hints and warnings

### ISSUE THREAD

**Issue by PatrickvL** [Cxbx-Reloaded/Cxbx-Reloaded#55]
Having no compiler hints and warnings gives a much cleaner build.

Causing new compiler messages should be forbidden - if any appear, the person responsible for that should fix it.

**Comment by x1nixmzeng** [comment]
Agreed, the flag to treat warnings as errors should be set, and perhaps raise the error level?

However, as there are copies of specific SDKs some files will flagged as deprecated, ie `GetVersionExA` in `multimon.h` from `import\DirectX8\include\multimon.h`

**Comment by x1nixmzeng** [comment]
Fixes from #100 have helped with this issue.

The biggest issue left is the alignment of XBE and EXE structures - as the use of `#include "AlignPosfix1.h"` is not really supported with other compilers.

Otherwise there are just lots of printf-style format specifiers which need updating!

**Comment by x1nixmzeng** [comment]
Ah, also the device enumeration callback issue which I don't know how to solve yet:

```
1>..\..\src\Common\Win32\XBController.cpp(933,28): error : no member named 'WrapEnumGameCtrlCallback' in 'XBController'
1>              &XBController::WrapEnumGameCtrlCallback,
1>               ~~~~~~~~~~~~~~^
1>..\..\src\Common\Win32\XBController.cpp(973,60): error : use of undeclared identifier 'WrapEnumObjectsCallback'
1>          m_InputDevice[m_dwCurObject].m_Device->EnumObjects(WrapEnumObjectsCallback, this, DIDFT_ALL);
1>                                                             ^
```

**Comment by Margen67** [comment]
If we're trying to get rid of warnings maybe the warning level should be raised?
![capture](https://user-images.githubusercontent.com/3462541/40161889-f4fb0b32-5966-11e8-871c-468dd7ab7899.PNG)

**Comment by PatrickvL** [comment]
Our own code should generate warnings, imported code shouldn't.

**Comment by PatrickvL** [comment]
In an ideal world, our code shouldn't raise any warning even when the compiler uses its most thorough validation methods. But right now it might not be feasible to reach that level of confirmation to the highest possible standards

**Comment by RadWolfie** [comment]
Moving comment to here from #1262 conversation.
>There's no way to treat warning as error level to be raise. The option is either treat warning as error or warning. See "Treat Warnings as Errors" option.
>
>Visual Studio's "Warning Level" is entirely different thing. It only verbose more warnings to alert the developers about something need to be fix before sending new release to public or private group.

And other comment from #1259
> Since we are "stuck" with deprecated GetVersion function from CxbxKrnl.cpp file. Plus other solution I discover are using Windows' API. We are not able to treat warning as error for time being.

Plus it is best to move part of the code for Operating System build check into their own directory, Win32 and Linux.
